pyctcdecode/decoder.py

```diff
@@ -71,6 +84,19 @@
 EMPTY_START_BEAM: Beam = ("", "", "", None, [], NULL_FRAMES, 0.0)
 
 
+# Generic float type
+if sys.version_info < (3, 8):
+    NpFloat = Any
+else:
+    if sys.version_info < (3, 9) and not TYPE_CHECKING:
+        NpFloat = Any
+    else:
+        NpFloat = np.floating[NBitBase]
+
+FloatVar = TypeVar("FloatVar", bound=NpFloat)
```

Comment:
Can we use `np.float_` here? https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy.float_

Reply:
my preference would be not to declare what kind of float the user is using, so float/double/long double should all be acceptable, which I think is the point of np.floating. The NBitBase helps to use the type hierarchy to ensure that information isn't lost in casts (i.e. float32 --> float64 is ok but not the opposite)

Reply:
Makes sense -- thanks